framework/behaviors/SluggableBehavior.php

```diff
@@ -171,6 +171,10 @@ protected function isNewSlugNeeded()
             return false;
         }
 
+        if ($this->attribute === null) {
```

Comment:
Do you think we should move this condition to a separate method?

I added a WIP comment. The problem is that it tells that `callback-based` slugs are always reevaluated but in fact all `value-based` slugs are always reevaluated at the moment.